dev/com.ibm.ws.jdbc_fat_postgresql/test-applications/postgresqlApp/src/jdbc/fat/postgresql/web/PostgreSQLTestServlet.java

```diff
@@ -217,6 +224,60 @@ public void testDefaultAutoCommit() throws Exception {
         }
     }
 
-    // TODO: Add a test for defaultAutoCommit setting in a global transaction
+    @Test
+    public void testDefaultAutoCommitOffGlobalTran() throws Exception {
+        // Ensure defaultAutoCommit=false behaves properly across global transaction boundaries.
+        // Insert/read data with two different DataSources, expect writes to auto-commit
+        DataSource regularDS = InitialContext.doLookup("jdbc/postgres/xa");
+        DataSource autoCommitDS = InitialContext.doLookup("jdbc/postgres/defaultAutoCommitOff");
+        try (Connection writingConn = autoCommitDS.getConnection();
+                        Connection readingConn = regularDS.getConnection()) {
+            assertFalse("DataSource with autoCommit=false set in server.xml should have autoCommit=false", writingConn.getAutoCommit());
+            // Insert and select some data to confirm the DB behaves as auto-commit=false
+            Statement insert = writingConn.createStatement();
+            insert.execute("INSERT INTO people(id,name) VALUES(14,'testDefaultAutoCommitOffGlobalTran')");
+
+            Statement query = readingConn.createStatement();
+            ResultSet rs = query.executeQuery("SELECT * FROM people WHERE id=14");
+            assertFalse("Should not find uncommitted row in database during global transaction", rs.next());
+
+            writingConn.commit();
+            rs = query.executeQuery("SELECT * FROM people WHERE id=14");
+            assertTrue("Did not find expected row in database after issuing a local Connection.commit()", rs.next());
+            assertEquals("testDefaultAutoCommitOffGlobalTran", rs.getString(2));
+        }
+
+        // Get a new connection, intentionally leave it open over a transaction boundary
+        Connection boundaryPassingConnection = autoCommitDS.getConnection();
+
+        tx.begin();
+        try (Connection writingConn = autoCommitDS.getConnection();
+                        Connection readingConn = regularDS.getConnection()) {
+            // Insert and select some data to confirm the DB behaves as auto-commit=false
+            Statement insert = writingConn.createStatement();
+            insert.execute("INSERT INTO people(id,name) VALUES(15,'testDefaultAutoCommitOffGlobalTran')");
+
+            Statement query = readingConn.createStatement();
+            ResultSet rs = query.executeQuery("SELECT * FROM people WHERE id=15");
+            assertFalse("Should not find uncommitted row in database during global transaction", rs.next());
+        }
+        tx.commit();
+
+        // Connection obtained before tx scope should have same autoCommit value
+        assertFalse("DataSource with autoCommit=false set in server.xml should have autoCommit=false", boundaryPassingConnection.getAutoCommit());
```

Comment:
This test would be more valuable if `boundaryPassingConnection` had enlisted in the global transaction by doing some work within it.